pkg/storage/provider_no_send_test.go

```diff
@@ -3,162 +3,378 @@ package storage_test
 import (
 	"testing"
 
-	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/internal/fixtures"
 	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/internal/fixtures/testusers"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/randomizer"
 	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/storage/internal/testabilities"
 	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/wdk"
-	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/wdk/primitives"
-	"github.com/go-softwarelab/common/pkg/slices"
-	"github.com/go-softwarelab/common/pkg/to"
 	"github.com/stretchr/testify/require"
 )
 
-func TestNoSendSendWithScenario_SendWithoutNewTx(t *testing.T) {
-	// given:
-	const inputSatoshis = 99904
+func TestNoSendPlusSendWithScenario(t *testing.T) {
+	t.Run("two no-send txs, many initial UTXOs, all noSendChange is used, no newTx when sendWith", func(t *testing.T) {
```

Comment:
Check this out, it looks a little bit weird, but maybe it's reading better 🤷 
Maybe this approach is worth to discuss on our slack channel.


```suggestion
	t.Run(`Alice has many UTXOs.
				She created two subsequent no-send transactions,
				and then send them in SendWith with not new transaction`,
		func(t *testing.T) {
```

And the result in console looks like that, IMO it's still quite readable:
```sh
=== RUN   TestNoSendPlusSendWithScenario/Alice_has_many_UTXOs._____She_created_two_subsequent_no-send_transactions,_____and_then_send_them_in_SendWith_with_not_new_transaction
```

Reply:
I would prefer describing details in a comment rather than in such complex test name

Reply:
you still have complex test name. But current name, doesn't mean much to me.

Reply:
The description is in code (self-explanatory) and additional comments. 
The test name shows which "case" is considered, like:
- how many transactions
- what are predefined UTXOs
- etc...

We can keep polishing the names still, but this won't be very productive and won't have real impact if code-reader "understand" the test or not.